### PR TITLE
Do not perform dependency scanning when no source inputs are specified

### PR DESCRIPTION
**Pull request by artemcm**
Resolves rdar://118900740

### CODE REVIEW

**Comment by artemcm** [comment]
@swift-ci test

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 1]
```diff
@@ -2924,7 +2933,7 @@ final class SwiftDriverTests: XCTestCase {
       throw StringError("Unexpected argument for output file map")
     }
     let firstKeyHandle = try XCTUnwrap(outFileMap.entries.keys.first)
-    let firstKey = try VirtualPath.lookup(firstKeyHandle).description
+    let firstKey = VirtualPath.lookup(firstKeyHandle).description
```
Comment (cachemeifyoucan) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:2936]: This warning fix should be a separate commit?

**Review by cachemeifyoucan — APPROVED** [review]
LGTM.

**Comment by artemcm** [comment]
@swift-ci test

**Comment by artemcm** [comment]
@swift-ci test Windows platform

**Comment by artemcm** [comment]
@swift-ci test Windows platform
